### PR TITLE
Use relative labels and `Label` macro wherever possible

### PR DESCRIPTION
**Pull request by UebelAndre**
This PR updates changes the explicit use of `@io_bazel_rules_rust//...` to `//...` and adds the use of [Label](https://docs.bazel.build/versions/master/skylark/lib/Label.html) wherever possible. This should currently have no functional but it provides some better type guarantees.

### CODE REVIEW

**Comment by UebelAndre** [comment]
This PR came from some exploration i was doing in trying to support either agnostic workspace names for this repository or at least both `rules_rust` and `io_bazel_rules_rust`. For the most part, this would work if it weren't for two things, the way toolchains are currently setup, and the fact that some `cargo-raze` outputs (namely protobuf) have dependencies on the workspace name.

File: bindgen/bindgen.bzl [hunk 1]
```diff
@@ -168,16 +168,16 @@ rust_bindgen = rule(
...
             executable = True,
             allow_single_file = True,
             cfg = "exec",
         ),
     },
     outputs = {"out": "%{name}.rs"},
     toolchains = [
-        "@io_bazel_rules_rust//bindgen:bindgen_toolchain",
-        "@io_bazel_rules_rust//rust:toolchain",
+        str(Label("//bindgen:bindgen_toolchain")),
```
Comment (UebelAndre) [inline bindgen/bindgen.bzl:179]: I'm shopping around for some thoughts here and generally more insight on [Label](https://docs.bazel.build/versions/master/skylark/lib/Label.html). I'm not sure if this is an appropriate use of `Label`. Anyone have any experiences like this they can share?

**Comment by UebelAndre** [comment]
@dfreese Hey, are you able to take a look at this 🙏

**Comment by UebelAndre** [comment]
@damienmg hey, you wouldn't als have time to look at this would you 😅 🙏

**Comment by UebelAndre** [comment]
@illicitonion Hey, would *you* have time to take a look at this? 🙏 😅

**Comment by UebelAndre** [comment]
@damienmg Hey, spoke with @illicitonion and he thinks you'd be the best reviewer for this change. Would you have a quick moment to take a look at this this week?

File: bindgen/bindgen.bzl (hunk 1 above)
Comment (damienmg) [reply bindgen/bindgen.bzl:179]: This is perfectly acceptable. I remember having a funny discussion with several senior engineer in the Bazel team about this "hack" and having the manager answer: I don't think this is a hack, which just settled our decision to keep this as an idiom, although I left the team 3 years ago so I don't know what is the status now.

**Review by damienmg — APPROVED** [review]

**Comment by damienmg** [comment]
This is the right approach, this repository was created before we made the label behavior consistent and thus was using aboslute label approach, but there is no reason to do this anymore.

**Comment by UebelAndre** [comment]
@damienmg Thanks for the review, that bit of context, and updating the branch 🙏  CI seems to still be passing 😄

**Pull request merged**